python/lib/core/dmod/core/dataset.py

```diff
@@ -267,6 +340,22 @@ def _set_expires(self, new_expires: datetime):
         self.expires = new_expires
         self.last_updated = datetime.now()
 
+    @property
+    def archive_name(self) -> Optional[str]:
```

Comment:
In the strictest sense, probably no.  But it seems intuitive to model it as a derivative property of a _Dataset_ instance, and trivial to implement.  Is there a reason why we shouldn't have it?

I did specifically separate the static method so that things could replicate a value without having a _Dataset_ object directly, assuming the hypothetical params could be determined, though I do see even that as being more appropriate if tied to the _Dataset_ class.